packages/fury-adapter-oas3-parser/lib/parser/oas/parseServerObject.js

```diff
@@ -15,6 +15,38 @@ const parseServerVariableObject = require('./parseServerVariableObject');
 const name = 'Server Object';
 const requiredKeys = ['url'];
 
+const validateVariableInURL = (context, object) => {
+  const url = object.getValue('url');
+  const variables = object.get('variables');
+  const parseResult = new context.namespace.elements.ParseResult();
+
+  const variablesKeys = variables.content.map(variable => variable.key.toValue());
+  const urlVariables = url
+    .match(/{(.*?)}/g)
+    .map(x => x.replace(/[{}]/g, ''));
```

Comment:
```suggestion
  const urlVariables = Array.from(url.matchAll(/{(.*?)}/g))
    .map(match => match[1]);
```

We could use [matchAll](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/matchAll) since that will return the capture groups so you can extract the value as-is instead of having to then perform a second regex query. It will likely be a bit more performant.

```js
> Array.from('{scheme}://{host}/'.matchAll(/{(.*?)}/g))
[
  [
    '{scheme}',
    'scheme',
    index: 0,
    input: '{scheme}://{host}/',
    groups: undefined
  ],
  [
    '{host}',
    'host',
    index: 11,
    input: '{scheme}://{host}/',
    groups: undefined
  ]
]
```

Reply:
I see the point, however `matchAll` is supported from Node 12 (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/matchAll#Browser_compatibility)

In the package.json it is specified that it should be used any Node version from 8 (https://github.com/apiaryio/api-elements.js/blob/10841673ef9cd813ac9c1db7a57c1bc71765f5a4/packages/fury-adapter-oas3-parser/package.json#L41)

I think due to the fact that Apiary Core still uses Node 10 most have set this as default.
What should I do then? We could even add an `.nvm` file so we set a version that we want.

For the moment I leave the function as it is

Reply:
Ok, then let's go with what you've got. Didn't know that wasn't available pre Node 12 when writing that.